Add U55 and U85 16A8W linear tests

Ninja91 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #14368

Add U55 and U85 16A8W quantization tests for linear operations following the pattern established in D80510628 for mul operations.

This adds test coverage for 16A8W (16-bit activations, 8-bit weights) quantization on both Ethos-U55 and Ethos-U85 platforms for linear operations. Both tests are marked with xfail annotations due to the known Vela compilation issue where int16 linear operations fail with "Conv2D Invalid arguments" error.

The tests follow the same structure as the existing 16A8W TOSA test but target specific hardware platforms:
- test_linear_16a8w_u55_INT16: Tests on U55 with rank1 data
- test_linear_16a8w_u85_INT16: Tests on U85 with both rank1 and rank4 data

Both tests use the same symmetric A16W8 quantization configuration and are properly decorated with platform-specific skip conditions (XfailIfNoCorstone300/320) and pytest xfail markers referencing GitHub issue 13947.

Addresses: https://github.com/pytorch/executorch/issues/13729

Differential Revision: [D82618039](https://our.internmc.facebook.com/intern/diff/D82618039/)

Ninja91 committed:
Add U55 and U85 16A8W linear tests

Add U55 and U85 16A8W quantization tests for linear operations following the pattern established in D80510628 for mul operations.

This adds test coverage for 16A8W (16-bit activations, 8-bit weights) quantization on both Ethos-U55 and Ethos-U85 platforms for linear operations. Both tests are marked with xfail annotations due to the known Vela compilation issue where int16 linear operations fail with "Conv2D Invalid arguments" error.

The tests follow the same structure as the existing 16A8W TOSA test but target specific hardware platforms:
- test_linear_16a8w_u55_INT16: Tests on U55 with rank1 data
- test_linear_16a8w_u85_INT16: Tests on U85 with both rank1 and rank4 data

Both tests use the same symmetric A16W8 quantization configuration and are properly decorated with platform-specific skip conditions (XfailIfNoCorstone300/320) and pytest xfail markers referencing GitHub issue 13947.

Addresses: https://github.com/pytorch/executorch/issues/13729

Differential Revision: [D82618039](https://our.internmc.facebook.com/intern/diff/D82618039/)

[ghstack-poisoned]

digantdesai commented on backends/arm/test/ops/test_linear.py:
```diff
@@ -308,3 +308,69 @@ def test_linear_16a8w_tosa_INT(test_data: torch.Tensor):
+
+@common.parametrize("test_data", test_data_rank1_INT | test_data_rank4_INT)
+@common.XfailIfNoCorstone320
+@pytest.mark.xfail(
+    reason="Vela compilation fails with 'Invalid arguments' for int16 linear operations. See: https://github.com/pytorch/executorch/issues/13947"
+)
```
Thanks Nitin, just making sure..

Ninja91 committed:
Update on "Add U55 and U85 16A8W linear tests"


Add U55 and U85 16A8W quantization tests for linear operations following the pattern established in D80510628 for mul operations.

This adds test coverage for 16A8W (16-bit activations, 8-bit weights) quantization on both Ethos-U55 and Ethos-U85 platforms for linear operations. Both tests are marked with xfail annotations due to the known Vela compilation issue where int16 linear operations fail with "Conv2D Invalid arguments" error.

The tests follow the same structure as the existing 16A8W TOSA test but target specific hardware platforms:
- test_linear_16a8w_u55_INT16: Tests on U55 with rank1 data
- test_linear_16a8w_u85_INT16: Tests on U85 with both rank1 and rank4 data

Both tests use the same symmetric A16W8 quantization configuration and are properly decorated with platform-specific skip conditions (XfailIfNoCorstone300/320) and pytest xfail markers referencing GitHub issue 13947.

Addresses: https://github.com/pytorch/executorch/issues/13729

Differential Revision: [D82618039](https://our.internmc.facebook.com/intern/diff/D82618039/)

[ghstack-poisoned]

Ninja91 committed:
Update on "Add U55 and U85 16A8W linear tests"


Add U55 and U85 16A8W quantization tests for linear operations following the pattern established in D80510628 for mul operations.

This adds test coverage for 16A8W (16-bit activations, 8-bit weights) quantization on both Ethos-U55 and Ethos-U85 platforms for linear operations. Both tests are marked with xfail annotations due to the known Vela compilation issue where int16 linear operations fail with "Conv2D Invalid arguments" error.

The tests follow the same structure as the existing 16A8W TOSA test but target specific hardware platforms:
- test_linear_16a8w_u55_INT16: Tests on U55 with rank1 data
- test_linear_16a8w_u85_INT16: Tests on U85 with both rank1 and rank4 data

Both tests use the same symmetric A16W8 quantization configuration and are properly decorated with platform-specific skip conditions (XfailIfNoCorstone300/320) and pytest xfail markers referencing GitHub issue 13947.

Addresses: https://github.com/pytorch/executorch/issues/13729

Differential Revision: [D82618039](https://our.internmc.facebook.com/intern/diff/D82618039/)

[ghstack-poisoned]

Ninja91 committed:
Update on "Add U55 and U85 16A8W linear tests"


Add U55 and U85 16A8W quantization tests for linear operations following the pattern established in D80510628 for mul operations.

This adds test coverage for 16A8W (16-bit activations, 8-bit weights) quantization on both Ethos-U55 and Ethos-U85 platforms for linear operations. Both tests are marked with xfail annotations due to the known Vela compilation issue where int16 linear operations fail with "Conv2D Invalid arguments" error.

The tests follow the same structure as the existing 16A8W TOSA test but target specific hardware platforms:
- test_linear_16a8w_u55_INT16: Tests on U55 with rank1 data
- test_linear_16a8w_u85_INT16: Tests on U85 with both rank1 and rank4 data

Both tests use the same symmetric A16W8 quantization configuration and are properly decorated with platform-specific skip conditions (XfailIfNoCorstone300/320) and pytest xfail markers referencing GitHub issue 13947.

Addresses: https://github.com/pytorch/executorch/issues/13729

Differential Revision: [D82618039](https://our.internmc.facebook.com/intern/diff/D82618039/)

[ghstack-poisoned]
